/*
 * @begin:license
 *
 * Copyright (c) 2015-2019, Benjamin Niemann <pink@odahoda.de>
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * @end:license
 */

syntax = "proto2";

package noisicaa.pb;

import "noisicaa/audioproc/public/node_parameters.proto";

message MidiCCtoCVChannelSpec {
  required uint32 midi_channel = 1;
  required uint32 midi_controller = 2;
  required uint32 initial_value = 3;
  required float min_value = 4;
  required float max_value = 5;
  required bool log_scale = 6;
}

message MidiCCtoCVSpec {
  repeated MidiCCtoCVChannelSpec channels = 1;
}

extend NodeParameters {
  optional MidiCCtoCVSpec midi_cc_to_cv_spec = 410000;
}
